client/client.go

```diff
@@ -7,21 +23,24 @@ import (
 )
 
 func main() {
-	secretprovider, err := sp.NewSecretProvider()
+	// Initializing secret provider
+	// Pre requisites and behavior are mentioned in READ me.
+	// First argument is the provider type, which can be sp.VPC/sp.Bluemix/sp.Softlayer
+	// Note: The first argument is of use, only when storage-secret-store is used (so that api key can be fetched either VPC or Bluemix or Softlayer section in slclient.toml)
+	secretprovider, err := sp.NewSecretProvider(sp.VPC)
+	// OR, you may also provide a key which is expected to be present the k8s secret - ibm-cloud-credentials/storage-secret-store
+	// secretprovider, err := sp.NewSecretProvider(sp.VPC, "your-key")
```

Comment:
The first argument is not applicable in case of passing a custom key so we can think of maybe making it an optional arg...

Reply:
I won't modify the function definition as of now. Will keep a note of this and take it up in upcoming PRs